Parsers/ASimAuthentication/Parsers/vimAuthenticationSalesforceSC.yaml

```diff
@@ -0,0 +1,364 @@
+        "n", "Chatter Communities External User Third Party SSO", "Other",
+        "r", "Employee Login to Community", "Other",
+        "z", "Lightning Login", "Username & Password",
+        "l", "Networks Portal API Only", "Other",
+        "6", "Remote Access Client", "Other",
+        "i", "Remote Access 2.0", "Other",
+        "I", "Other Apex API", "Other",
+        "R", "Partner Product", "Other",
+        "w", "Passwordless Login", "Passwordless",
+        "3", "Customer Service Portal", "Other",
+        "q", "Partner Portal Third-Party SSO", "Other",
+        "9", "Partner Portal", "Other",
+        "5", "SAML Idp Initiated SSO", "Other",
+        "m", "SAML Chatter Communities External User SSO", "Other",
+        "b", "SAML Customer Service Portal SSO", "Other",
+        "c", "SAML Partner Portal SSO", "Other",
+        "h", "SAML Site SSO", "Other",
+        "8", "SAML Sfdc Initiated SSO", "Other",
+        "E", "SelfService", "Other",
+        "j", "Third Party SSO", "Other"
+    ];
+      let LogonProtocolLookup = datatable(
+        LoginSubType_s: string,
+        LogonProtocolOriginal: string,
+        LogonProtocol: string
+    )[
+        "uiup", "UI Username-Password", "Basic Auth",
+        "oauthpassword", "OAuth Username-Password", "OAuth",
+        "oauthtoken", "OAuth User-Agent", "OAuth",
+        "oauthhybridtoken", "OAuth User-Agent for Hybrid Apps", "OAuth",
+        "oauthtokenidtoken", "OAuth User-Agent with ID Token", "OAuth",
+        "oauthclientcredential", "OAuth Client Credential", "OAuth",
+        "oauthcode", "OAuth Web Server", "OAuth",
+        "oauthhybridauthcode", "OAuth Web Server for Hybrid Apps", "OAuth",
+    ];
+      let TempEventResultLookup = datatable(request_status_s: string, TempEventResult: string)[
+      "S", "Success",
+      "F", "Failure",
+      "A", "Failure",
+      "R", "Success",
+      "N", "Failure",
+      "U", "NA"
+  ];
+      let UserTypeLookup = datatable(user_type_s: string, TargetUserType: string)[
+      "CsnOnly", "Other",
+      "CspLitePortal", "Other",
+      "CustomerSuccess", "Other",
+      "Guest", "Anonymous",
+      "PowerCustomerSuccess", "Other",
+      "PowerPartner", "Other",
+      "SelfService", "Other",
+      "Standard", "Regular",
+      "A", "Application",
+      "b", "Other",
+      "C", "Other",
+      "D", "Other",
+      "F", "Other",
+      "G", "Anonymous",
+      "L", "Other",
+      "N", "Service",
+      "n", "Other",
+      "O", "Other",
+      "o", "Other",
+      "P", "Other",
+      "p", "Other",
+      "S", "Regular",
+      "X", "Admin"
+  ];
+      union isfuzzy=true
+        SalesforceSchema,
+        SalesforceServiceCloud_CL 
+      | where not(disabled)
+      //  -- Pre filtering
+      | where 
+          (isnull(starttime) or todatetime(tostring(split(timestamp_s, '.', 0)[0])) >= starttime) 
+          and (isnull(endtime) or todatetime(tostring(split(timestamp_s, '.', 0)[0])) <= endtime) 
```

Comment:
computing it several times. Performance wise, is this better than using extended and run the filter on it ?